server/repository/src/migrations/v1_06_00/stock_line_add_supplier_link_id.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{migrations::sql, StorageConnection};
+
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    #[cfg(feature = "postgres")]
+    sql!(
+        connection,
+        r#"
+        -- Adding stock_line.supplier_link_id
+        ALTER TABLE stock_line
+        ADD COLUMN supplier_link_id TEXT;
+        
+        UPDATE stock_line
+        SET supplier_link_id = supplier_id;
+        
+        ALTER TABLE stock_line ADD CONSTRAINT stock_line_supplier_link_id_fkey FOREIGN KEY (supplier_link_id) REFERENCES name_link(id);
+        "#,
+    )?;
+
+    #[cfg(not(feature = "postgres"))]
+    sql!(
+        connection,
+        r#"
+        -- Adding stock_line.supplier_link_id
+        PRAGMA foreign_keys = OFF;
+        ALTER TABLE stock_line
+        ADD COLUMN supplier_link_id TEXT REFERENCES name_link(id);
+        
+        UPDATE stock_line
+        SET supplier_link_id = supplier_id;
+        PRAGMA foreign_keys = ON;
+     "#,
+    )?;
+
+    sql! {
+        connection,
+        r#"
+        DROP INDEX index_stock_line_supplier_id;
```

Comment:
I tweaked this just removing the IF EXISTS, think it's straight forward enough I'll do it in the other PRs that I requested the same. Test pass!

That said, I see that at least for this one the index is created in 1_06_00 migration as well (maybe why you changed the order of where the indexing migration is in 1_06_00). Maybe we should just remove it rather than drop it straight after creating it.

Reply:
If the merging goes into 1.6 then we can... but looks like they'll be going to 1.7 at this rate?